### ISSUE TITLE
Check for optional types fails with newer Python union-typing syntax

### ISSUE THREAD

**Issue by markhepburn** [kevinsbarnard/ifdo-py#2]
In ifdo/model.py#L44, the origin check fails because newer versions of python report differently depending on whether the type was declared as `Union[str,None]` (Or `Optional[str]`), or the newer `str|None`.

To verify:
```
>>> from typing import get_origin, Union
>>> get_origin(Union[str,None])
typing.Union
>>> get_origin(str|None)
<class 'types.UnionType'>
```

I am using python 3.12, but I believe that syntax was introduced in 3.10.

**Comment by cjackett** [comment]
Hi Mark,

Thanks for raising this compatibility issue. We've implemented a fix that handles both `Union[str, None]` and the newer `str | None` syntax. This is now available in v1.2.2 on PyPI: https://pypi.org/project/ifdo/1.2.2/

Please let us know if you encounter any other issues.

**Comment by markhepburn** [comment]
Thanks for a speedy response @cjackett !
